### PR TITLE
use native schema serializer

### PR DESCRIPTION
**Pull request by MichaelBelousov**
- native schema deserialization breaks on custom attributes that are not defined before their usage, which happens when serializing the biscore schema with the ecschema-metadata xml serializer, as a work around, we invoke the native serializer instead
  - add an end-to-end transformation test that upgrading biscore works, which fails with the current ecschema-metadata serializer
- change confusing `--targetFile` cli arg which always deletes the target when working with local files to treat the target as the transform target, and instead a new cli arg, `--targetDestination` which always overwrites the file at that destination

### CODE REVIEW

**Comment by MichaelBelousov** [comment]
I'm going to make this target master instead and backport it, but I wasn't able to test it on master (AuthType doesn't exist at runtime and a type error is thrown)

**Comment by MichaelBelousov** [comment]
@Mergifyio backport release/2.19.x

**Comment by MichaelBelousov** [comment]
If anyone received an email about being added as a reviewer, I'm sorry, I rebased. Next time I will close and open a new PR before rebasing.

**Review by mgooding — APPROVED** [review]
Approved, but would like a sanity check from @ColinKerr.

**Review by ColinKerr — APPROVED** [review]

File: core/backend/src/IModelTransformer.ts [hunk 1]
```diff
@@ -766,11 +766,19 @@ export class IModelTransformer extends IModelExportHandler {
...
 
+  private _hasNativelyExportedAllSchemas = false;
+
   /** Override of [IModelExportHandler.onExportSchema]($backend) that serializes a schema to disk for [[processSchemas]] to import into
    * the target iModel when it is exported from the source iModel. */
-  protected override async onExportSchema(schema: ECSchemaMetaData.Schema): Promise<void> {
-    const schemaPath = path.join(this._schemaExportDir, `${schema.fullName}.ecschema.xml`);
-    IModelJsFs.writeFileSync(schemaPath, await schema.toXmlString());
+  protected override async onExportSchema(_schema: ECSchemaMetaData.Schema): Promise<void> {
+    // HACK: a bug in the native deserializer means we can't yet always read schemas serialized from this end
```
Comment (calebmshafer) [inline core/backend/src/IModelTransformer.ts:774]: Is there a fix in the works for the native side?

File: core/backend/src/IModelTransformer.ts [hunk 2]
```diff
@@ -766,11 +766,19 @@ export class IModelTransformer extends IModelExportHandler {
...
+    // so we invoke the native serializer as a workaround for now
+    if (!this._hasNativelyExportedAllSchemas) {
+      this._hasNativelyExportedAllSchemas = true;
+      this.targetDb.nativeDb.exportSchemas(this._schemaExportDir);
+    }
+    // const schemaPath = path.join(this._schemaExportDir, `${schema.fullName}.ecschema.xml`);
```
Comment (calebmshafer) [inline core/backend/src/IModelTransformer.ts:780]: Mind deleting the commented out code?

**Review by calebmshafer — APPROVED** [review]

File: core/backend/src/IModelTransformer.ts (hunk 1 above)
Comment (MichaelBelousov) [reply core/backend/src/IModelTransformer.ts:774]: yes I submitted a native PR and will submit another one here reverting this change with the test passing.

File: core/backend/src/IModelTransformer.ts [hunk 3]
```diff
@@ -966,16 +973,16 @@ export class TemplateModelCloner extends IModelTransformer {
     targetElementProps.federationGuid = Guid.createValue(); // clone from template should create a new federationGuid
     targetElementProps.code = Code.createEmpty(); // clone from template should not maintain codes
     if (sourceElement instanceof GeometricElement3d) {
-      const placement = Placement3d.fromJSON((targetElementProps as GeometricElement3dProps).placement);
```
Comment (MichaelBelousov) [inline core/backend/src/IModelTransformer.ts]: this appears to have been accidentally changed somehow, reverting

Comment (MichaelBelousov) [reply core/backend/src/IModelTransformer.ts]: looks like my linter got confused somehow and autofixed on me, reverting

**Comment by MichaelBelousov** [comment]
@calebmshafer getting flaky builds, including 
-  full stack tests database operation failure:
  ```sh
    1) iModelHub iModelsHandler
       should filter iModels by template:
     iModelHub.DatabaseOperationFailed: Database operation has failed.
      at parse (/home/vsts/work/1/s/clients/imodelhub/src/imodelhub/Errors.ts:93:21)
      at Object.request (/home/vsts/work/1/s/clients/itwin/src/Request.ts:438:25)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at IModelBaseHandler.postInstance (/home/vsts/work/1/s/clients/itwin/src/WsgClient.ts:319:27)
      at IModelsHandler.createIModelInstance (/home/vsts/work/1/s/clients/imodelhub/src/imodelhub/iModels.ts:480:16)
      at IModelsHandler.create (/home/vsts/work/1/s/clients/imodelhub/src/imodelhub/iModels.ts:649:20)
      at Context.<anonymous> (src/imodelhub/iModels.test.ts:847:37)
  ```
- windows rush cover failed:
  ```sh
   Error: Command failed: node D:\vsts_a\672\s\tools\build\scripts\test.js --grep "#integration|#WebGLPerformance" --offline mock --invert
    at checkExecSyncError (child_process.js:760:11)
    at Object.execSync (child_process.js:833:15)
    at exec (D:\vsts_a\672\s\tools\build\bin\betools.js:218:24)
    at testCommand (D:\vsts_a\672\s\tools\build\bin\betools.js:166:3)
    at Object.handler (D:\vsts_a\672\s\tools\build\bin\betools.js:52:17)
    at Object.runCommand (D:\vsts_a\672\s\common\temp\node_modules\.pnpm\yargs@16.2.0\node_modules\yargs\build\index.cjs:446:48)
    at Object.parseArgs [as _parseArgs] (D:\vsts_a\672\s\common\temp\node_modules\.pnpm\yargs@16.2.0\node_modules\yargs\build\index.cjs:2697:57)
    at Object.get [as argv] (D:\vsts_a\672\s\common\temp\node_modules\.pnpm\yargs@16.2.0\node_modules\yargs\build\index.cjs:2651:25)
    at Object.<anonymous> (D:\vsts_a\672\s\tools\build\bin\betools.js:151:3)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
  npm ERR! Test failed.  See above for more details.
  ```
- a timeout on the docs build

**Comment by calebmshafer** [comment]
@Mergifyio update

**Comment by calebmshafer** [comment]
@MichaelBelousov usually the iModelHub one happens when we overload it with too many PR builds and restarting it in a few minutes normally helps.

The `rush cover` one is something we thought we fixed with an update to the Node version however seems to be persistent but just happening less often.  I'll add an item to the list to dig into it.

The docs build will expire after 3 hours if the PR is out of date with master. It actually will mark any of the status checks as invalid, just depends on the list.

**Comment by MichaelBelousov** [comment]
I misread the doc build error, sorry, I thought it was saying the build minion lost contact and timed out after 3 hours. This was the 4th or so flaky failure which is why I added a comment. I'll just keep re-queuing builds slowly until it gets past the flakiness

**Comment by calebmshafer** [comment]
@MichaelBelousov ah okay, let me know if it persists. I'll dig into it more.

**Comment by MichaelBelousov** [comment]
@Mergifyio update

File: core/backend/src/test/standalone/IModelTransformer.test.ts [hunk 4]
```diff
@@ -1130,4 +1130,18 @@ describe("IModelTransformer", () => {
     targetDb.close();
   });
 
+  it("biscore update is valid", async () => {
```
Comment (calebmshafer) [inline core/backend/src/test/standalone/IModelTransformer.test.ts:1133]: @MichaelBelousov have you tried skipping this test to see if you can get the backend ones to pass? I haven't been seeing any other PRs with such an ability to reproduce the error you're seeing here.

Comment (MichaelBelousov) [reply core/backend/src/test/standalone/IModelTransformer.test.ts:1133]: I haven't tried doing that to see if the CI passes, but I will. This test shouldn't be doing that much new compared to other tests we already have here.

**Comment by MichaelBelousov** [comment]
@mergifyio update

**Pull request merged**